cmd/vulcan-prowler/main.go

```diff
@@ -0,0 +1,286 @@
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/iam"
+
+	check "github.com/adevinta/vulcan-check-sdk"
+	"github.com/adevinta/vulcan-check-sdk/state"
+	report "github.com/adevinta/vulcan-report"
+	"github.com/aws/aws-sdk-go/aws/arn"
+)
+
+const (
+	defaultRegion          = `eu-west-1`
+	defaultSessionDuration = 3600 // 1 hour.
+
+	envEndpoint = `VULCAN_ASSUME_ROLE_ENDPOINT`
+	envRole     = `ROLE_NAME`
+
+	envKeyID     = `AWS_ACCESS_KEY_ID`
+	envKeySecret = `AWS_SECRET_ACCESS_KEY`
+	envToken     = `AWS_SESSION_TOKEN`
+)
+
+var (
+	checkName = "vulcan-prowler"
+	logger    = check.NewCheckLog(checkName)
+
+	defaultGroups = []string{
+		"cislevel2",
+	}
+
+	CISCompliance = report.Vulnerability{
+		Summary: "Compliance With CIS AWS Foundations Benchmark (BETA)",
+		Description: `<p>
+			The CIS AWS Foundations Benchmark provides prescriptive
+			guidance for configuring security options for a subset of Amazon Web
+			Services with an emphasis on foundational, testable, and architecture
+			agnostic settings. The services included in the scope are:
+		</p>
+		<p>
+		<ul>
+			<li>IAM</li>
+			<li>Config</li>
+			<li>CloudTrail</li>
+			<li>CloudWatch</li>
+			<li>SNS</li>
+			<li>S3</li>
+			<li>VPC (Default)</li>
+		</ul>
+		</p>
+		<p>
+			The provided recommendations are classified in 2 different levels: 1
+			and 2, being the level 2 intended for environments or use cases where
+			security is paramount.
+		</p>
+		<p>
+			Check the Details and Resources sections to know the compliance status
+			and more details.
+		</p>`,
+		References: []string{
+			"https://d0.awsstatic.com/whitepapers/compliance/AWS_CIS_Foundations_Benchmark.pdf",
+			"https://github.com/toniblyx/prowler",
+			"https://www.cisecurity.org/benchmark/amazon_web_services/",
+		},
+	}
+)
+
+type options struct {
+	Region          string   `json:"region"`
+	Groups          []string `json:"groups"`
+	SessionDuration int      `json:"sessionDuration"` // In secs.
+}
+
+func buildOptions(optJSON string) (options, error) {
+	var opts options
+	if optJSON != "" {
+		if err := json.Unmarshal([]byte(optJSON), &opts); err != nil {
+			return opts, err
+		}
+	}
+	if opts.Region == "" {
+		opts.Region = defaultRegion
+	}
+	if opts.Groups == nil {
+		opts.Groups = defaultGroups
+	}
+	if opts.SessionDuration == 0 {
+		opts.SessionDuration = defaultSessionDuration
+	}
+
+	return opts, nil
+}
+
+func main() {
+	run := func(ctx context.Context, target string, optJSON string, state state.State) error {
+		if target == "" {
+			return errors.New("check target missing")
+		}
+		parsedARN, err := arn.Parse(target)
+		if err != nil {
+			return err
+		}
+
+		opts, err := buildOptions(optJSON)
+		if err != nil {
+			return err
+		}
+
+		endpoint := os.Getenv(envEndpoint)
+		role := os.Getenv(envRole)
+
+		logger.Infof("using endpoint '%s' and role '%s'", endpoint, role)
+
+		if err := loadCredentials(endpoint, parsedARN.AccountID, role, opts.SessionDuration); err != nil {
```

Comment:
I think it's not strictly required but, maybe it would be good checking that envRole and envEndpoint are at least not empty before calling loadCredentials.

Reply:
The role is optional, but I will add check for the endpoint. Thanks!

Reply:
Done